rpc/core/net.go

```diff
@@ -105,15 +107,37 @@ func (env *Environment) UnsafeDialPeers(
 // Genesis returns genesis file.
 // More: https://docs.cometbft.com/main/rpc/#/Info/genesis
 func (env *Environment) Genesis(*rpctypes.Context) (*ctypes.ResultGenesis, error) {
-	if len(env.genChunks) > 1 {
+	if len(env.genChunks) > 0 {
 		return nil, ErrGenesisRespSize
 	}
 
 	return &ctypes.ResultGenesis{Genesis: env.GenDoc}, nil
 }
 
-func (env *Environment) GenesisChunked(_ *rpctypes.Context, chunk uint) (*ctypes.ResultGenesisChunk, error) {
+func (env *Environment) GenesisChunked(
+	_ *rpctypes.Context,
+	chunk uint,
+) (*ctypes.ResultGenesisChunk, error) {
 	if env.genChunks == nil {
+		// See discussion in the following PR for why we still serve chunk 0 even
+		// if env.genChunks is nil:
+		// https://github.com/cometbft/cometbft/pull/4235#issuecomment-2389109521
+		if chunk == 0 {
```

Comment:
To my understanding this is , from the clients perspective, the same behaviour. For both endpoints.

Reply:
It is not new behavior. The current implementation of `InitGenesisChunks()` creates the `genChunks` slice even when the genesis's size is <= `genesisChunkSize` (i.e., chunks aren't needed). In this case, the `genChunks` slice has only 1 element at index 0: the entire genesis encoded as a base64 string. If you query the `/genesis_chunked` endpoint with chunk id = 0, it will return you the entire genesis encoded as base64.

The code you are seeing here preserves this behavior after our changes.

Reply:
sounds good, thanks for the clarification